### PR TITLE
Fix patch version of our dependency on syn.

### LINKED ISSUE #3151: `pyo3-macros-backend` using `syn` 1.0.85 method, but depends on 1.0.85
### Bug Description

(I'm still a relative newbie to the Rust ecosystem, so sorry if this isn't a problem or is a problem with my build)

I'm attempting to upgrade PyO3 to 0.18(.4) using `pyo3 =  { git = "https://github.com/PyO3/pyo3.git", rev = "c7d541b9ad2a9ef0255ab69541e67b22fe3fbf32"}` (thanks for fixing https://github.com/PyO3/pyo3/issues/3138) and get the following:

```
error[E0599]: no method named `token` found for reference `&LitStr` in the current scope
   --> /home/josh/.cargo/git/checkouts/pyo3-a22e69bc62b9f0fd/c7d541b/pyo3-macros-backend/src/pyfunction/signature.rs:595:57
    |
595 |                         syn::Lit::Str(s) => default = s.token().to_string(),
    |                                                         ^^^^^ method not found in `&LitStr`
```
I see that:
- `ListStr.token()` was added in `syn` 1.85: [1.84 docs](https://docs.rs/syn/1.0.84/syn/struct.LitStr.html) [1.85 docs](https://docs.rs/syn/1.0.85/syn/struct.LitStr.html)
- The `syn` dependency of `pyo3-macros-backend` [uses version](https://github.com/PyO3/pyo3/blob/c7d541b9ad2a9ef0255ab69541e67b22fe3fbf32/pyo3-macros-backend/Cargo.toml#L21) `1.0.56`

So I suspect the dependency is incorrect?

### Steps to Reproduce

I think if you make a crate that just depends on `pyo3-macros-backend` and declares a `pyfunction` to might see the error?
Sorry, I see it in a complex project :sweat_smile: 

### Backtrace

_No response_

### Your operating system and version

N/A

### Your Python version (`python --version`)

N/A

### Your Rust version (`rustc --version`)

rustc 1.69.0 (84c898d65 2023-04-16)

### Your PyO3 version

c7d541b9ad2a9ef0255ab69541e67b22fe3fbf32

### How did you install python? Did you use a virtualenv?

N/A

### Additional Info

:pray: Thanks again

### PR DESCRIPTION
**Pull request by adamreichold**
Closes #3151

### CODE REVIEW

File: newsfragments/3152.fixed.md
Comment (davidhewitt) [inline newsfragments/3152.fixed.md:1]: Maybe this should be 3152.packaging.md ?

**Review by davidhewitt — APPROVED** [review]
Thanks! Just one suggestion.

File: newsfragments/3152.fixed.md
Comment (adamreichold) [reply newsfragments/3152.fixed.md:1]: Certainly, I was going "label: bug! ... category: fix!" and I did not even consider alternatives.

**Comment by adamreichold** [comment]
bors r=davidhewitt

**Comment by adamreichold** [comment]
bors r=davidhewitt

**Comment by adamreichold** [comment]
bors r=davidhewitt

**Pull request merged**